src/providers/ibmcloud/mod.rs

```diff
@@ -112,6 +126,29 @@ impl IBMGen2Provider {
         }
         output
     }
+
+    /// Find the SSH keys in the vendordata file
+    fn fetch_ssh_keys(vendordata_vec: Vec<u8>) -> Result<Vec<String>> {
+        // Parse MIME format from vendor-data file
+        let vendor_data_mail = parse_mail(&vendordata_vec).unwrap();
+        let mut cloud_config = String::new();
+        for section in vendor_data_mail.subparts {
+            for header in &section.headers {
+                if let "text/cloud-config" = header.get_value().as_str() {
+                    if section.get_body().unwrap().contains("ssh_authorized_keys") {
```

Comment:
I think you can replace this `unwrap` with an `unwrap_or_default`.

Reply:
Changed in the next commit.